File: android/src/main/kotlin/dev/steenbakker/flutter_ble_peripheral/models/PeripheralState.kt
package dev.steenbakker.flutter_ble_peripheral.models

/**
 * Represents the various states of the Flutter BLE Peripheral.
 */
enum class PeripheralState {

    /**
     * The current BLE status is not yet determined.
     */
    unknown,

    /**
     * BLE is not supported on this device.
     */
    unsupported,

    /**
     * The app is not authorized to use BLE.
     */
    unauthorized,

    /**
     * Bluetooth is currently turned off.
     */
    poweredOff,

    /**
     * Android only: Location services are disabled, which may affect BLE functionality.
     */
    locationServicesDisabled,

    /**
     * BLE is available and ready to use, but not currently advertising or connected.
     */
    idle,

    /**
     * BLE is actively advertising data.
     */
    advertising,

    /**
     * BLE is connected to a remote device.
     */
    connected
}

File: darwin/flutter_ble_peripheral/Sources/flutter_ble_peripheral/Models/PeripheralState.swift
//
//  PeripheralStatus.swift
//  flutter_ble_peripheral
//
//  Created by Julian Steenbakker on 26/11/2021.
//

import Foundation

/**
 Represents the current state of the Flutter BLE Peripheral.

 This enum mirrors the BLE state used on the Flutter side,
 providing a unified model for cross-platform BLE peripheral state management.
 */
enum PeripheralState: Int {

    /// The current BLE status is not yet determined.
    case unknown

    /// BLE is not supported on this device.
    case unsupported

    /// The app is not authorized to use BLE.
    case unauthorized

    /// Bluetooth is currently turned off.
    case poweredOff
    
    /// Android only: Location services are disabled.
    case locationServicesDisabled

    /// BLE is available and ready to use, but not currently advertising or connected.
    case idle

    /// BLE is actively advertising data.
    case advertising

    /// BLE is connected to a remote device.
    case connected
}
